internal/client/cloud_client.go

```diff
@@ -91,8 +91,52 @@ func (c *CloudClient) Do(req *http.Request) (*ResponseWithETag, error) {
 		return nil, err
 	}
 
-	// Extract the ETag if it exists
-	etag := resp.Header.Get("ETag")
+	// Extract the ETag - check multiple possible header names
+	var etag string
+
+	// Check standard ETag header (case-insensitive)
+	etag = resp.Header.Get("ETag")
+
+	// If not found, check for other variations
+	if etag == "" {
+		etag = resp.Header.Get("etag")
+	}
+	if etag == "" {
+		etag = resp.Header.Get("Etag")
+	}
+
+	// Sometimes ETags are returned with quotes, clean those if present
+	if len(etag) > 0 && (etag[0] == '"' && etag[len(etag)-1] == '"') {
+		etag = etag[1 : len(etag)-1]
+	}
+
+	// If ETag is still not found, try to extract it from the response body
+	if etag == "" && (resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusCreated) {
+		// Only attempt this for successful responses
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err == nil {
+			// Create a copy of the body for later use
+			resp.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+
+			// Try to find ETag in a common field pattern
+			var respMap map[string]interface{}
+			if err := json.Unmarshal(bodyBytes, &respMap); err == nil {
+				// Check for common ETag field patterns in the response
+				if etagVal, ok := respMap["etag"].(string); ok {
+					etag = etagVal
+				} else if etagVal, ok := respMap["ETag"].(string); ok {
+					etag = etagVal
+				} else if etagVal, ok := respMap["e_tag"].(string); ok {
+					etag = etagVal
+				} else if etagVal, ok := respMap["_etag"].(string); ok {
+					etag = etagVal
+				} else if etagVal, ok := respMap["ConfigETag"].(string); ok {
+					// This is the format used by the AuthZed API
+					etag = etagVal
```

Comment:
This also seems like something that we should fix upstream if the header isn't being set rather than looking for it in the body. I don't think etags in the body are standard.